File: 02. C# Part2/03. Methods-Homework/06. FirstLargerThanNeighbours/FirstLargerThanNeighbours.cs
using System;
using System.Linq;

        //Write a method that returns the index of the first element in array that is larger 
        //than its neighbours, or -1, if there’s no such element.
        //Use the method from the previous exercise.

    class FirstLargerThanNeighbours
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Enter the integers of the array: ");
            int[] numbers = Console.ReadLine()
                .Split(new char[] { ' ', ',', '\t' }, StringSplitOptions.RemoveEmptyEntries)
                .Select(x => int.Parse(x))
                .ToArray();
            int firstLarger = FirstLarger(numbers);
            Console.WriteLine("The first larger number is: {0}", firstLarger);
        }

        private static int FirstLarger(int[] numbers)
        {
            for (int i = 1; i < numbers.Length -1; i++)
            {
                if (numbers[i ] > numbers[i + 1] && numbers[i ] > numbers[i + 1])
                {
                    return numbers[i];
                }
            }
            return -1;
        }
    }

File: 06. JavaScript Basics/06. Functions-Homework/06. Functions-Homework/07. FirstLargerThanNeighbours/FirstLargerThanNeighbours.js
//Write a function that returns the index of the first element in array that is larger than its neighbours or -1, 
//if there’s no such element.
//Use the function from the previous exercise.

function largerThanNeighbours(arr) {
    var found;
    for (var i = 0; i < arr.length; i++) {
        if (arr[i] > arr[i - 1] && arr[i] > arr[i + 1]) {
            jsConsole.writeLine("In the array [" + arr + "] The first found element is " + arr[i] + " with an index: " + i);
            found = true;
            break;
        }
    }
    if (!found) {
       jsConsole.writeLine("There is no such element.");
    }
}

largerThanNeighbours([1, 2, 8, 6, 89, 2, 5, 3, 1]);
largerThanNeighbours([1, 1, 1, 1, 1]);